web3js-ext/test/role_based_key_feepayer.spec.ts

```diff
@@ -0,0 +1,141 @@
+                    { type: AccountKeyType.Public, key: pub1 },
+                    { type: AccountKeyType.Public, key: pub2 },
+                    { type: AccountKeyType.Public, key: pub3 }
+                ]
+            }
+        };
+
+        const signedUpdateTx = await roleAccountUpdate.signTransaction(updateTx);
+        console.log("Account Updated:", signedUpdateTx);
+        assert.isNotNull(signedUpdateTx.transactionHash, "Account update transaction should succeed");
+    });
+
+    // Test Case 1: Sending a normal transaction with RoleTransaction key
+    it("1. Sending a normal transaction with RoleTransaction key", async function () {
+        const valueTx = {
+            type: TxType.ValueTransfer,
+            from: generateTemporaryAccount().address,
+            to: generateTemporaryAccount().address,
+            value: toPeb("0.01"),
+            gasLimit: 100000
+        };
+
+        const signedTx = await roleTransactionAccount.signTransaction(valueTx);
+        console.log("RoleTransaction signedTx:", signedTx.transactionHash);
+        assert.isNotNull(signedTx.transactionHash, "RoleTransaction transaction should succeed");
+    });
+
+    // Test Case 2: Attempting to sign a regular transaction with RoleFeePayer key (should fail)
+    it("2. Attempting to sign a regular transaction with RoleFeePayer key (failure test)", async function () {
+        const valueTx = {
+            type: TxType.ValueTransfer,
+            from: generateTemporaryAccount().address,
+            to: generateTemporaryAccount().address,
+            value: toPeb("0.01"),
+            gasLimit: 100000
+        };
+
+        try {
+            const signedTx = await roleFeePayerAccount.signTransaction(valueTx);
+            console.log("Unexpected Success - Signed Transaction:", signedTx);
+            assert.fail("RoleFeePayer key should not sign a ValueTransfer transaction.");
+        } catch (error: any) {
+            console.log("Expected Error (RoleFeePayer):", error.message);
+            assert.isTrue(true, "Error occurred as expected due to role mismatch");
+        }
+    });
+
+    // Test Case 3: Fee Delegated transaction signed by RoleFeePayer key (should succeed)
+    it("3. Fee Delegated transaction signed by RoleTransaction and RoleFeePayer keys", async function () {
+        const feeDelegatedTx = {
+            type: TxType.FeeDelegatedValueTransfer,
+            from: generateTemporaryAccount().address,
+            to: generateTemporaryAccount().address,
+            value: toPeb("0.01"),
+            gasLimit: 100000
+        };
+
+        // Feedback2. Remove the dependencies with network connection. just check the signed result.
+        // Feedback3. signed by a user (from address,roleTransactionAccount) first
+        const signedTxBySender = await roleTransactionAccount.signTransaction(feeDelegatedTx);
+        assert.isNotNull(signedTxBySender.rawTransaction, "Transaction should be signed by RoleTransaction");
```

Comment:
[[Feedback 2]](https://github.com/kaiachain/kaia-sdk/pull/59/files#r1843200864) don't send the tx to the testnet, just check the signed result.
cc. @kjeom